Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -247,24 +247,31 @@ void CDMInstanceOpenCDM::requestLicense(LicenseType licenseType, const AtomicStr
         mimeType = "video/x-h264";
     else if (GStreamerEMEUtilities::isWidevineKeySystem(m_keySystem))
         mimeType = "video/mp4";
-    m_openCdmSession->CreateSession(mimeType.utf8().data(), reinterpret_cast<unsigned char*>(const_cast<char*>(initData->data())),
+
+    GST_TRACE("Going to request a new session ID");
+
+    bool createdSession = m_openCdmSession->CreateSession(mimeType.utf8().data(), reinterpret_cast<unsigned char*>(const_cast<char*>(initData->data())),
         initData->size(), sessionId, webKitLicenseTypeToOpenCDM(licenseType));
-    if (!sessionId.size()) {
+
+    if (!createdSession) {
         callback(WTFMove(initData), sessionIdValue, false, Failed);
         return;
     }
+
     sessionIdValue = String::fromUTF8(sessionId.c_str());
+    GST_TRACE("create session succeeded, session ID is %s", sessionIdValue.utf8().data());
 
-    unsigned char temporaryUrl[1024] = {'\0'};
+    unsigned char temporaryUrl[1024] = { };
     std::string message;
     int messageLength = 0;
     int destinationUrlLength = 0;
-    int returnValue = m_openCdmSession->GetKeyMessage(message,
-        &messageLength, temporaryUrl, &destinationUrlLength);
-    if (returnValue || !messageLength || !destinationUrlLength) {
+    m_openCdmSession->GetKeyMessage(message, &messageLength, temporaryUrl, &destinationUrlLength);
```

Comment:
I assume you give up the return value here for a good reason, right?

Reply:
It's always true...